warehouse/ingest-core/src/main/java/datawave/ingest/table/aggregator/GlobalIndexUidAggregator.java

```diff
@@ -172,48 +169,49 @@ public Value reduce(Key key, Iterator<Value> iter) {
                     for (String uid : v.getUIDList()) {
                         
                         // check that a removal has not occurred
-                        // if it has, we decrement the count, from above.
                         if (!uidsToRemove.contains(uid) && !quarantinedIds.contains(uid)) {
                             
                             // add the UID iff we are under our MAX
                             if (uids.size() < maxUids)
                                 uids.add(uid);
-                            
                         }
-                        
                     }
                     
-                    if (log.isDebugEnabled())
-                        log.debug("Adding uids " + delta + " " + count);
+                    // A partial major compaction could lead to a protocol buffer that has a positive count as well as
+                    // UIDs in the REMOVEDUID list. If we're encountering such a key here, then we need to be sure to
+                    // add those removed UIDs back in to the uidsToRemove list and not just drop them.
+                    if (propogate) {
+                        for (String uid : v.getREMOVEDUIDList()) {
+                            if (!uids.contains(uid))
+                                uidsToRemove.add(uid);
+                        }
+                    }
+                    
+                    log.debug("Adding uids {} {}", delta, count);
                     
                     // if our delta is < 0, then we can remove, iff seenIgnore is false. If it is true, there is no need to proceed with removals
-                } else if (delta < 0 && !seenIgnore) {
+                } else if (!seenIgnore) {
                     
                     // so that we can perform the decrement
                     for (String uid : v.getREMOVEDUIDList()) {
```

Comment:
again, not fully understanding our use of this protobuf, but in the above block we don't appear to care about the REMOVEDUIDList when propogate is false but here we're updating uidsToRemove regardless of the propagate setting. Does that make sense?

Reply:
If the propogate [sic] var is false, then removals aren't passed along (basically propogate is false for a full major compaction only). So yeah, we should probably be checking the propogate var here too. The seenIgnore flag gets set when there's a protocol buffer whose count exceeds the max. In that case, there's nothing in either the UID or REMOVEDUID list, so we should probably be checking that in the add case as well. I'm not an expert on this class and was just trying to fix a specific error that I encountered. I'm hesitant to make changes that are too sweeping, but if there are other experts on this class who want to weigh in and comment to ensure I haven't broken anything, that would be good. Errors in this class could have some pretty serious negative consequences.